### PR TITLE
sql: allow CREATE EXTENSION "uuid-ossp"

### LINKED ISSUE #80118: sql: support functions from uuid-ossp extension
Postgres has the `uuid-ossp` extension which provides builtin functions related to UUID generation: https://www.postgresql.org/docs/current/uuid-ossp.html

We already support the [`uuid_generate_v4`](https://github.com/cockroachdb/cockroach/blob/273cb8f7847625da0cf70d5697dbc61178a7ad5a/pkg/sql/sem/builtins/builtins.go#L683) function. We should be able to implement the others without much effort.

Once we do that, we can treat the `uuid-ossp` extension as a no-op, similar to how we treat the `postgis` extension. https://github.com/cockroachdb/cockroach/blob/273cb8f7847625da0cf70d5697dbc61178a7ad5a/pkg/sql/create_extension.go#L46

### Functions we need:

- [ ] uuid_generate_v1 () → uuid                 Generates a version 1 UUID. This involves the MAC  address of the computer and a time stamp. Note that UUIDs of this kind  reveal the identity of the computer that created the identifier and the  time at which it did so, which might make it unsuitable for certain  security-sensitive applications.

- [ ] uuid_generate_v1mc () → uuid                 Generates a version 1 UUID, but uses a random multicast MAC address instead of the real MAC address of the computer.

- [ ] uuid_generate_v3 ( namespace uuid, name text ) → uuid                 Generates a version 3 UUID in the given namespace  using the specified input name. The namespace should be one of the  special constants produced by the uuid_ns_*() functions. The generation of  UUIDs by this method has no random or environment-dependent element and  is therefore reproducible.

- [ ] uuid_generate_v4 () → uuid                 Generates a version 4 UUID, which is derived entirely from random numbers.

- [ ] uuid_generate_v5 ( namespace uuid, name text ) → uuid                 Generates a version 5 UUID, which works like a  version 3 UUID except that SHA-1 is used as a hashing method. Version 5  should be preferred over version 3 because SHA-1 is thought to be more  secure than MD5.

- [ ] uuid_nil () → uuid  Returns a “nil” UUID constant, which does not occur as a real UUID.

- [ ] uuid_ns_dns () → uuid  Returns a constant designating the DNS namespace for UUIDs.

- [ ] uuid_ns_url () → uuid  Returns a constant designating the URL namespace for UUIDs.

- [ ] uuid_ns_oid () → uuid  Returns a constant designating the ISO object identifier (OID) namespace for UUIDs. (This pertains to ASN.1 OIDs, which are unrelated to the OIDs used in PostgreSQL.)

- [ ] uuid_ns_x500 () → uuid  Returns a constant designating the X.500 distinguished name (DN) namespace for UUIDs.

Epic CRDB-14998

### PR DESCRIPTION
**Pull request by rafiss**
fixes #80118

Release note (sql change): Added the builtin functions: uuid_nil, uuid_ns_dns,
uuid_ns_url, uuid_ns_oid, and uuid_ns_x500.

Release note (sql change): The builtin functions uuid_generate_v1,
uuid_generate_v1mc, uuid_generate_v3, and uuid_generate_v5 were added.

Release note (sql change): The command CREATE EXTENSION "uuid-ossp" no
longer fails, since CockroachDB now includes all the builtin functions from
this extension.

### CODE REVIEW

File: pkg/util/uuid/generator.go [hunk 1]
```diff
@@ -285,3 +285,16 @@ func defaultHWAddrFunc() (net.HardwareAddr, error) {
 	}
 	return []byte{}, fmt.Errorf("uuid: no HW address found")
 }
+
+// RandomHardwareAddrFunc rReturns a random hardware address, with the multicast
```
Comment (ajwerner) [inline pkg/util/uuid/generator.go:289]: nit: `rReturns`

**Review by ajwerner — APPROVED** [review]
LGTM

**Review by otan — APPROVED** [review]

**Comment by rafiss** [comment]
bors r=otan,ajwerner

**Comment by pschichtel** [comment]
@rafiss has this already been part of a release?

**Comment by rafiss** [comment]
This will be in the v22.2.0 release, scheduled for early December. There are already beta versions and release candidates available for download.

**Pull request merged**